File: Tutorial1-1Basics/src/main/java/com/example/roompersistencelibrary/data/UserDao.java
package com.example.roompersistencelibrary.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

/**
 * This interface is used for CRUD operation on database
 */
@Dao
public interface
UserDao {

    @Insert
    public long insertUser(User user);

    @Query("SELECT * FROM users")
    public List<User> getAll();

    @Delete
    public void delete(User user);

}

File: Tutorial1-1Basics/src/main/java/com/example/roompersistencelibrary/data/UserDao.kt
package com.example.roompersistencelibrary.data

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

/**
 * This interface is used for CRUD operation on database
 */
@Dao
interface UserDao {
    @Insert
    fun insertUser(user: User?): Long

    @get:Query("SELECT * FROM users")
    val all: List<User?>?

    @Delete
    fun delete(user: User?)
}